File: Library.WebUI/src/app/models/book/bookUpdateView.ts
import { AuthorGetView } from "../author/authorGetView";
import { PublicationHouseGetView } from "../publicationHouse/publicationHouseGetView";

export class BookUpdateView {
  constructor(
    public id?: number,
    public name?: string,
    public yearOfPublication?: number,

    public author?: AuthorGetView,
    public publicationHouses?: PublicationHouseGetView[]) { }
}

File: Library.ViewModels/Book/BookUpdateView.cs
﻿using Library.ViewModels.PublicationHouse;
using System.Collections.Generic;

namespace Library.ViewModels.Book
{
    public class BookUpdateView
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublication { get; set; }
        public int? AuthorId { get; set; }
		public List<int> PublicationHouseIds { get; set; }
    }
}
